compiler/AST/CatchStmt.cpp

```diff
@@ -243,13 +258,23 @@ void CatchStmt::cleanup()
 
   newBody->insertAtHead(castedDef);
 
-  if (catchall) {
-    VarSymbol* error = new VarSymbol(name);
+  VarSymbol* error = NULL;
+  DefExpr* errorDef = NULL;
+  if (_name != NULL) {
+    // Grab the VarSymbol we created during ::build()
+    // We 'remove()' the Def to insert it in the newBody with an init-expr.
+    errorDef = toDefExpr(oldBody->body.head->remove());
+    error = toVarSymbol(errorDef->sym);
```

Comment:
I think it'd be nice to have an INT_ASSERT here checking that `errorDef->sym->name == _name`